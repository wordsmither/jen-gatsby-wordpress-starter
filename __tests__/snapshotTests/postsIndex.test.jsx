import { render } from '@testing-library/react'

import PostIndexTemplate from '../../src/templates/postsIndex'
import data from '../data/allWpPost.json'

describe('<PostIndexTemplate />', () => {
	beforeEach(() => {
		location = {
			pathname: '/posts/1',
			state: { breakOpen: false },
		}
	})

	it('should render with posts', () => {
		const { asFragment } = render(
			<PostIndexTemplate
				location={location}
				pageContext={{
					itemsPerPage: 2,
					routing: true,
					posts: data,
				}}
			/>,
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
