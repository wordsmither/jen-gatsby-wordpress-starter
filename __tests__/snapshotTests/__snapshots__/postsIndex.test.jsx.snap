// Vitest Snapshot v1

exports[`<PostIndexTemplate /> > should render with posts 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen max-h-screen min-w-screen max-w-screen flex flex-col"
  >
    <div
      class="my-0 pt-10 px-5 text-xl"
    >
      <nav>
        <ul
          class="flex flex-row flex-wrap sm:flex-nowrap list-none justify-between max-w-screen-sm mx-auto"
        >
          <li
            class="mr-auto"
          >
            <a
              class="font-sans hover:underline"
              href="/"
            >
              üè† Home
            </a>
          </li>
          <li
            class="mx-4"
          >
            <a
              class="font-sans hover:underline"
              href="/posts"
            >
              üì∞ Posts
            </a>
          </li>
          <li
            class="mx-4"
          >
            <a
              class="font-sans hover:underline"
              href="/pages"
            >
              üìë Pages
            </a>
          </li>
          <li
            class="mx-4"
          >
            <a
              class="font-sans hover:underline"
              href="/examples"
            >
              ‚öõÔ∏è Examples
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <main
      class="mb-auto"
    >
      <header
        class="prose text-2xl mx-auto mt-20"
      >
        <h1
          class="text-center mx-auto"
        >
          Posts
        </h1>
      </header>
      <div
        class="max-w-screen-lg mx-auto"
      >
        <section>
          <div
            class="max-w-full"
          >
            <span
              class="mb-8 prose-sm font-bold"
            >
              Page 1/1
            </span>
            <section>
              <div
                class="mt-12 grid gap-5 max-w-content mx-auto lg:grid-cols-3 lg:max-w-screen-lg"
              >
                <a
                  href="/posts/example-post-with-image/"
                >
                  <div
                    class="flex flex-col rounded-lg shadow-lg overflow-hidden cursor-pointer border-2 h-full hover:border-indigo-500"
                  >
                    <div
                      class="flex-shrink-0 relative h-40"
                    >
                      <div
                        class="gatsby-image-wrapper gatsby-image-wrapper-constrained h-full w-full"
                        data-gatsby-image-wrapper=""
                        style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
                      >
                        <div
                          style="max-width: 640px; display: block;"
                        >
                          <img
                            alt=""
                            aria-hidden="true"
                            role="presentation"
                            src="data:image/svg+xml;charset=utf-8,%3Csvg height='360' width='640' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                            style="max-width: 100%; display: block; position: static;"
                          />
                        </div>
                        <div
                          aria-hidden="true"
                          data-placeholder-image=""
                          style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                        />
                        <picture>
                          <source
                            data-srcset="/static/045d1d479eed7ff0a5984b59c75d98b9/c0299/pizza.webp 160w,/static/045d1d479eed7ff0a5984b59c75d98b9/378b7/pizza.webp 320w,/static/045d1d479eed7ff0a5984b59c75d98b9/3c3fb/pizza.webp 640w"
                            sizes="(min-width: 640px) 640px, 100vw"
                            type="image/webp"
                          />
                          <img
                            alt="Example Post with Image"
                            data-gatsby-image-ssr=""
                            data-main-image=""
                            data-src="/static/045d1d479eed7ff0a5984b59c75d98b9/0f1c0/pizza.jpg"
                            data-srcset="/static/045d1d479eed7ff0a5984b59c75d98b9/c898a/pizza.jpg 160w,/static/045d1d479eed7ff0a5984b59c75d98b9/2278c/pizza.jpg 320w,/static/045d1d479eed7ff0a5984b59c75d98b9/0f1c0/pizza.jpg 640w"
                            decoding="async"
                            loading="lazy"
                            sizes="(min-width: 640px) 640px, 100vw"
                            style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; object-fit: fit; opacity: 0;"
                          />
                        </picture>
                        <noscript />
                        <script
                          type="module"
                        >
                          const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
                        </script>
                      </div>
                    </div>
                    <h2
                      class="my-4 mx-6 text-xl leading-7 font-semibold text-gray-900"
                    >
                      Example Post with Image ‚Üí
                    </h2>
                  </div>
                </a>
                <a
                  href="/posts/hello-world/"
                >
                  <div
                    class="flex flex-col rounded-lg shadow-lg overflow-hidden cursor-pointer border-2 h-full hover:border-indigo-500"
                  >
                    <div
                      class="flex-shrink-0 relative h-40"
                    >
                      <div
                        class="w-full h-full bg-gradient-to-b from-blue-100 to-blue-500"
                      />
                    </div>
                    <h2
                      class="my-4 mx-6 text-xl leading-7 font-semibold text-gray-900"
                    >
                      Hello world! ‚Üí
                    </h2>
                  </div>
                </a>
              </div>
            </section>
            <div
              class="sticky lg:bottom-12 bottom-4 mt-10"
            />
          </div>
        </section>
      </div>
    </main>
    <footer
      class="w-full text-white bg-black p-4 mt-12"
    >
      <nav
        class="flex flex-col max-w-lg mx-auto lg:max-w-screen-lg"
      >
        <ul>
          <li
            class="ml-3 list-disc text-blue-300 hover:text-blue-100"
          >
            <a
              class="text-blue-300 hover:text-blue-100 focus:text-purple-600 active:text-purple-300 "
              href="/posts/example-post-with-image/"
            >
              Example Post with Image
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="flex my-4 p-2"
      >
        <span
          class="mx-auto"
        >
          ¬© 2023 Built with 
          <a
            class="text-purple-300 underline hover:text-blue-100"
            href="https://www.gatsbyjs.com"
          >
            Gatsby
          </a>
           and 
          <a
            class="text-blue-200 underline hover:text-blue-100"
            href="https://wordpress.org/"
          >
            WordPress
          </a>
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
